### DISCUSSION TITLE
d5.11.4

### DISCUSSION THREAD

**Discussion by ghost** [modula3/cm3#780] [General]
See [Getting Started](https://github.com/modula3/cm3/wiki/Getting-Started) for help building and installing this release.

<hr /><em>This discussion was created from the release <a href='https://github.com/modula3/cm3/releases/tag/d5.11.4'>d5.11.4</a>.</em>

**Comment by jaykrell** [comment]
Thank you so much for this!
This is a *major* advance in ease of acquisition, building nicely on the C++ backend, and the refactoring of m3core/unix.

We might want to rename AMD64_LINUX to Unix, and "hide" I386_LINUX somewhere like "alternatives".
Similarly AMD64_NT is "Windows" and I386_NT is "alternatives".

Merging these into one download is difficult and easy at the same time.

If nobody complains about (doubling) download size, each .cpp file can be ifdef win32 ... else ... endif, and union the CMakeLists.txt.
Most of the files will be identical, so the ifdef provider could be smarter.

m3build could "compile" everything unconditionally, but put ifdefs around -- which works when there is FooWin.m3 and FooUnix.m3, but not when there are two Foo.m3. ifdefs could be for endian too.

**Reply by MarcusE1W** [reply]
That is a good idea. I used AMD64 to build cm3 for aarch64 (ARM) on Linux without problem. The package is more universal than it’s name suggests

**Reply by tkurtbond** [reply]
When I tried that on Ubuntu 22.04 on my Raspberry Pi 400, it told me that:
    AARCH64_LINUX is not a recognized target
Any ideas?

**Reply by MarcusE1W** [reply]
Sorry for the late reply.
I had the same problem but in this issue a solution is given
#805 
I have installed it on a Pinebook with Manjaro

**Comment by kwolcott** [comment]
Any possibility that this could be built from source more easily for MacOS arm64?  I don't wish to use MacOS Rosetta to run Intel/AMD 64 bit apps (I try to have only arm64 binaries/app on my machine), which looks like it is the only way to go for now.

Thanks,
Ken Wolcott
